src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/gitlab/GitlabServerPullRequestDecorator.java

```diff
@@ -91,117 +98,213 @@ public GitlabServerPullRequestDecorator(Server server, ConfigurationRepository c
-                            .filter(i -> i.hasChangesetForLine(issue.getIssue().getLine()))
-                            .map(i -> i.getChangesetForLine(issue.getIssue().getLine()))
-                            .map(Changeset::getRevision)
-                            .filter(commits::contains)
-                            .isPresent()) {
-                        //only if the change is on a commit, that belongs to this MR
-
-                        List<NameValuePair> fileContentParams = Arrays.asList(
-                                new BasicNameValuePair("body", fileComment),
-                                new BasicNameValuePair("position[base_sha]", mergeRequest.getDiffRefs().getBaseSha()),
-                                new BasicNameValuePair("position[start_sha]", mergeRequest.getDiffRefs().getStartSha()),
-                                new BasicNameValuePair("position[head_sha]", mergeRequest.getDiffRefs().getHeadSha()),
-                                new BasicNameValuePair("position[old_path]", path),
-                                new BasicNameValuePair("position[new_path]", path),
-                                new BasicNameValuePair("position[new_line]", String.valueOf(issue.getIssue().getLine())),
-                                new BasicNameValuePair("position[position_type]", "text"));
-
-                        postCommitComment(mergeRequestDiscussionURL, headers, fileContentParams, fileCommentEnabled);
-                    } else {
-                        LOGGER.info(String.format("Skipping %s:%d since the commit does not belong to the MR", path, issue.getIssue().getLine()));
-                    }
-                }
+            doPipeline(analysis);
+            doSummary(analysis, summaryComment);
+            doFileComments(analysis, lineComments);
+            if (deleteCommentsEnabled) {
+	            for( Note note: lineComments.values()) {
+	            	deleteCommitDiscussionNote(pullRequestId, note.getDiscussionId(), note.getId());            	
+	            }
             }
         } catch (IOException ex) {
             throw new IllegalStateException("Could not decorate Pull Request on Gitlab Server", ex);
         }
 
     }
 
-    private <X> X getSingle(String userURL, Map<String, String> headers, Class<X> type) throws IOException {
-        HttpGet httpGet = new HttpGet(userURL);
+
+	protected void doFileComments(AnalysisDetails analysis, Map<String, Note> lineNotes) throws IOException {
+		// Issues
+        final boolean fileCommentEnabled = getMandatoryBooleanProperty(PULL_REQUEST_FILE_COMMENT_ENABLED);
+        final boolean compactCommentsEnabled = getMandatoryBooleanProperty(PULL_REQUEST_COMPACT_COMMENTS_ENABLED);
+		if (fileCommentEnabled) {
+			Map<String, String> headers = getHeaders();
+	        final String pullRequestId = analysis.getBranchName();
+            final String mergeRequestDiscussionURL = getMergeRequestDiskussionsURL(pullRequestId);
```

Comment:
This is the german spelling, it should use a `c` instead of a `k`. There are more variables and Methodnames with this typo.